controllers/storagecluster/generate.go

```diff
@@ -79,7 +80,23 @@ func generateNameForSnapshotClassDriver(initData *ocsv1.StorageCluster, snapshot
 	return fmt.Sprintf("%s.%s.csi.ceph.com", initData.Namespace, snapshotType)
 }
 
-func generateNameForSnapshotClassSecret(snapshotType SnapshotterType) string {
+func generateNameForSnapshotClassSecret(instance *ocsv1.StorageCluster, snapshotType SnapshotterType) string {
+	if instance.Spec.ExternalStorage.Enable {
+		data, ok := externalOCSResources[instance.UID]
+		if !ok {
+			log.Error(fmt.Errorf("Unable to retrieve external resource from externalOCSResources"),
+				"unable to generate name for snapshot class secret for external mode")
+		}
+		// print the Secret name which contains the prefix as the rook-csi-rbd/cephfs-provisioner default secret name
+		// for example if the secret name is rook-csi-rbd-node-rookStorage-replicapool it will check the prefix with rook-csi-rbd-node if it matches it will return that name
+		for _, d := range data {
+			if d.Kind == "Secret" {
+				if strings.Contains(d.Name, fmt.Sprintf("rook-csi-%s-provisioner", snapshotType)) {
```

Comment:
what about the upgraded cluster? do we need to delete and recreate the snapshotclass?

Reply:
We are not supporting the Restricted modes for the upgraded cluster, will only be used for fresh deployments,

We can think about it for upgrades later too but for that, we need to create fresh CSI secrets too,
And ig for now any changes in the storage class are picked by the ocs-operator by deleting the class and creating it again. Not sure the same is implemented for volume snapshot class too or not, @iamniting

Reply:
I tested and even the upgrade was taken over by ocs-operator to update the volume-snapshot class

Reply:
No, it won't.  these secrets are generated by the python script what Rook generates. and we don't use newVolumeSnapshotClass or generateNameForSnapshotClassSecret in consumer/provider mode. but as you mentioned well its good to test these changes to make sure nothing breaks.

Reply:
So these secrets are not also coming from our API server in consumer mode? 
If not, what are the equivalent secrets in our case?

Reply:
No, in our case when we create the ceph users rook internally creates a secretes per user and extracts information from it, and returns secret name and data in Apiserver response https://github.com/red-hat-storage/ocs-operator/blob/main/services/provider/server/server.go#L565-L591

Reply:
right, thank you for clarifying